api/types/container/host_config.go

```diff
@@ -403,6 +403,7 @@ type HostConfig struct {
 	// Applicable to UNIX platforms
 	CapAdd          strslice.StrSlice // List of kernel capabilities to add to the container
 	CapDrop         strslice.StrSlice // List of kernel capabilities to remove from the container
+	Capabilities    []string          `json:",omitempty"` // Capabilities for the container
```

Comment:
I don't think we should be adding this field.
The default capabilities should only need to be set on the daemon and pass that down during OCI spec generation.

Reply:
Ok so where should I set these defaults?

Reply:
Defaults only need to be set on the daemon config.

Currently the default spec is generated in the OCI package and caps are set `DefaultLinuxSpec`. The list is hardcoded.

One approach would be to update `daemon.createSpec` to change the generated default spec with the ones in the daemon config. This means we are doing some allocations for the default list that we won't need anyway... but it's fairly minor and there's much worse things happening in that regard.

https://github.com/moby/moby/blob/46cdcd206c56172b95ba5c77b827a722dab426c5/daemon/oci_linux.go#L1012

So likely the simplest solution is:

```go
if len(daemon.config.DefaultCapabilities) > 0 {
	// copy to oci spec
}
```

Reply:
@cpuguy83  Take a look and see if this was what you where thinking when you get the chance.